### PR TITLE
Add retrying failed callback requests

### PR DESCRIPTION
**Pull request by bewakes**

### CODE REVIEW

File: analysis_module/models.py [hunk 1]
```diff
@@ -26,3 +34,59 @@ class FeaturesType(models.TextChoices):
...
+    status = models.PositiveIntegerField(
+        choices=FailedStatus.choices,
+        default=FailedStatus.RETRYING,
+    )
+
+    class Meta:
+        db_table = "failed_callback_tracker"
+
+    def resend_request(self):
+        if self.retries_count >= 5:
```
Comment (ranjan-stha) [inline analysis_module/models.py]: instead of hardcode, let's pass this as env var.

File: analysis_module/views.py [hunk 2]
```diff
@@ -16,18 +16,18 @@
...
+        response, code = topicmodeling_model(request)
     elif type == "summarization":
-        response, code = summarizationmodel(request)
+        response, code = summarization_model(request)
     elif type == "ngrams":
-        response, code = ngramsmodel(request)
+        response, code = ngrams_model(request)
     elif type == "geolocation":
-        response, code = geolocationmodel(request)
+        response, code = geolocation_model(request)
```
Comment (ranjan-stha) [inline analysis_module/views.py]: Can we do something like by avoiding if .. else....

action = {
    "topicmodel": topicmodeling_model,
    "summarization": summarization_model,
    "ngrams": ngrams_model,
    "geolocation": geolocation_model
    }
    
response, code = action.get(type)(request)

Something like this.

File: analysis_module/models.py (hunk 1 above)
Comment (bewakes) [reply analysis_module/models.py]: Done.

File: analysis_module/views.py (hunk 2 above)
Comment (bewakes) [reply analysis_module/views.py]: Done.

**Review by k9845 — APPROVED** [review]
LGTM

File: analysis_module/models.py [hunk 3]
```diff
@@ -26,3 +35,59 @@ class FeaturesType(models.TextChoices):
 
     class Meta:
         db_table = "event_status_tracker"
+
+
+class FailedCallback(BaseModel):
+    class FailedStatus(models.IntegerChoices):
```
Comment (thenav56) [inline analysis_module/models.py]: Maybe use different name :D 
```suggestion
class CallbackTrack(BaseModel):
    class Status(models.IntegerChoices):
```

**Review by thenav56 — COMMENTED** [review]
LGTM

File: analysis_module/models.py (hunk 3 above)
Comment (bewakes) [reply analysis_module/models.py]: FailedStatus -> Status sounds good and updated. The model name seems good to me. :D

**Review by thenav56 — APPROVED** [review]

**Review by ranjan-stha — APPROVED** [review]

**Pull request merged**